### PR TITLE
Add sticky functionality

### PR DESCRIPTION
**Pull request by vedhant**
Implements #51 .

### CODE REVIEW

**Comment by vedhant** [comment]
In #53 anything can be made stickable (even 3d objects). Its like calling `push()` and `pop()`. This will also help users to drag objects very easily. Although dropping could be an issue.
The following snippet illustrates it :
```javascript
let pg;
let randomx, randomy;
let dragging = false;

function preload() {
  createVRCanvas();
}

function setup() {
  setVRBackgroundColor(200, 0, 150);
  randomx = random(-500, 500);
  randomy = random(-500, 500);
}

function draw() {
  setViewerPosition(0, 0, 400);
  push();
  translate(randomx, randomy);
  if(dragging || intersectsSphere(70, 0, 0)) {
    dragging = true;
    sticky();
    fill('blue');
    sphere(70);
    noSticky();
  }
  else {
    fill('red');
    sphere(70);
  }
  pop();
}

```

File: src/p5xr/core/p5xrviewer.js [hunk 1]
```diff
@@ -48,3 +54,19 @@ p5.prototype.setViewerPosition = function(x, y, z) {
   let viewer = p5xr.instance.viewer;
   viewer.setPosition(x, y, z);
 };
+
+p5.prototype.sticky = function(drawOnTop = false) {
+  push();
+  p5xr.instance.viewer.drawOnTop = drawOnTop;
+  if(drawOnTop)
```
Comment (stalgiag) [inline src/p5xr/core/p5xrviewer.js]: I am not sure if I agree with keeping this as a property on the viewer. Seems a bit odd to have a `drawOnTop` property on a viewer. The conceptual link between a viewer and the order of rendering is a little tenuous. There is very little cost to enabling DEPTH_TEST with `noSticky` every time it is called.

File: src/p5xr/core/p5xrviewer.js [hunk 2]
```diff
@@ -48,3 +54,19 @@ p5.prototype.setViewerPosition = function(x, y, z) {
...
+    p5.instance._renderer.GL.disable(p5.instance._renderer.GL.DEPTH_TEST);
+  p5.instance._renderer.uMVMatrix.set(p5.Matrix.identity());
+  let viewerPosition = p5xr.instance.viewer.position;
+  setViewerPosition(viewerPosition.x, viewerPosition.y, viewerPosition.z);
+};
+
+p5.prototype.noSticky = function() {
```
Comment (stalgiag) [inline src/p5xr/core/p5xrviewer.js]: The use of `push` and `pop` here might warrant some safe guards. What happens if a user calls `sticky` but not `noSticky` in the draw loop? The model transforms will reset at the start of the loop but the state stack will not. Perhaps there needs to be some property on p5xr that knows whether the mode is `sticky` and `pop` that state at the end of the draw loop?

Comment (vedhant) [reply src/p5xr/core/p5xrviewer.js]: The model transforms will be reset anyways in the next draw loop. But there might be other properties which will not be restored. How can we detect end of draw loop? Perhaps users should make sure they call `noSticky()` appropriately?

Comment (stalgiag) [reply src/p5xr/core/p5xrviewer.js]: I think that is okay for the time being.

**Pull request merged**